src/utils/fp.tsx

```diff
@@ -1,22 +1,27 @@
+import type { ResolvedDeck } from "@/store/lib/types";
 import type { Card } from "@/store/services/queries.types";
 
-export type Filter = (c: Card) => boolean;
+type Filterable = Card | ResolvedDeck;
```

Comment:
Do we need the constraint? Shouldn't `T` suffice since these functions don't rely on any shared property of the type they act upon.

Reply:
I tried a bunch of things, and finally settled on this. A looser constraint also works, but you end up having to go and explicit the type everywhere else (so `Filter<Card>`, and `and<ResolvedDeck>`, and such). 
This way is a bit more verbose here, but I don't have to touch everywhere else. 

That I missed something completely is also a possibility, and we can do further testing ~

Reply:
Curious that typescript can't infer T from the argument, that should work🤔

Reply:
But yeah, let's leave it as is, it's fine